File: php/examples/example_exceptions.php
<?php
include __DIR__ . '/../VesselFinderApi.php';

use API\VesselFinderApi;
use Exceptions\ApiErrorException;
use Exceptions\ApiRequestErrorException;
use Exceptions\ApiInvalidArgumentsException;

$client = new VesselFinderApi('-- Input your userkey here --');

try {
    $vessels = $client->vessels([9228801,9441271], 227441980);
} catch (ApiInvalidArgumentsException $e) {
    echo sprintf("This block is for Invalid Arguments: %s", $e->getMessage()) . PHP_EOL;
} catch (ApiRequestErrorException $e) {
    echo sprintf("This block is for Request Error: %s", $e->getMessage()) . PHP_EOL;
} catch (ApiErrorException $e) {
    echo sprintf("General error: %s", $e->getMessage()) . PHP_EOL;
}

File: python/examples/example_exceptions.py
from vesselfinder_api import VesselFinderApi
from vesselfinder_api.exceptions import ApiErrorException, ApiInvalidArgumentsException, ApiRequestErrorException

v = VesselFinderApi(userkey='-- Input your userkey here --', errormode=False, save_last_info=True)

try:
    print(v.vessels(imo=[9228801,9441271], mmsi=227441980))
except ApiInvalidArgumentsException as e:
    print("This block is for Invalid Arguments: {}".format(e))
except ApiRequestErrorException as e:
    print("This block is for Request Error: {}".format(e))
except ApiErrorException as e:
    print("General case: {}".format(e))
